lib/puppet/indirector/facts/facter.rb

```diff
@@ -21,6 +21,13 @@ def save(facts)
   # Lookup a host's facts up in Facter.
   def find(request)
     Facter.reset
+
+    # Add the puppetversion fact; this is done before generating the hash so it is
+    # accessible to custom facts.
+    Facter.add(:puppetversion) do
+      setcode { Puppet.version.to_s }
+    end
+
```

Comment:
Re the facts in `add_local_facts` let's treat those as a separate concern. I created https://tickets.puppetlabs.com/browse/PUP-4457 for that, and we can revisit at a later date.

Meanwhile this looks good.